### PR TITLE
Re-enable mio-ffmpeg external build, now with CMake

### PR DESCRIPTION
**Pull request by tbabiin**
### Linked issues

Fixes SG-29155

### Summarize your change.

Adding MIO-FFMPEG external build code which was removed previously.

### Describe the reason for the change.

It is added for parity

### Describe what you have tested and on which operating system.

Linux Rocky 8, Win 11 & Win 10, MacOS 13.2.1

### Add a list of changes, and note any that might need special attention during the review.

Copying many files (mostly includes but some libraries) in the regular (internal) MIO_FFMPEG build to setup the external build.
Adding during Install (rvinst) that the files are enabled to be moved.
Configuring Windows as a Multi-Config project thus using CMake's Generator Expression to use Build-Time decisions.
Adding many find_library calls for loading libraries during the external build.
Note: To build externally a Debug version, Debug RV must be built.

### If possible, provide screenshots.

### CODE REVIEW

File: src/lib/image/mio_ffmpeg/ext_build/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+  ${_sources}
+)
+
+IF(APPLE)
+  # Darwin is the name of the mach BSD-base kernel :-)
+  SET(MIO_TARGET_DARWIN
+      BOOL TRUE "Detected target is Apple's macOS"
+      CACHE INTERNAL ""
+  )
+  TARGET_COMPILE_OPTIONS(
```
[the remark this answers is not in the document]
Comment (tbabiin) [reply src/lib/image/mio_ffmpeg/ext_build/CMakeLists.txt:51]: They, the compile options, aren't aligned actually. The external build needs a lot less options, meaning that the current list could be curated more if we wanted to spend more time.

**Comment by geffrak** [comment]
Can you add a ctest that runs the mio-ffmpeg build to make sure we don't break it?

**Comment by tbabiin** [comment]
> Can you add a ctest that runs the mio-ffmpeg build to make sure we don't break it?

I'll look, as look as the test can run during/after the build, it'll work.

**Review by bernie-laberge — APPROVED** [review]

**Comment by tbabiin** [comment]
@geffrak The test is ready for you to review. I'll merge sometime this afternoon if no new comments are added to this PR.

File: src/lib/image/mio_ffmpeg/CMakeLists.txt [hunk 2]
```diff
@@ -29,4 +32,278 @@ IF(RV_TARGET_WINDOWS)
...
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${IMAGE_MOVIE_FFMPEG_HEADERS} ${STAGE_MOVIE_FFMPEG_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${BASE_STL_EXT_HEADERS} ${STAGE_STL_EXT_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${IMAGE_TWEAK_TWKFB_HEADERS} ${STAGE_TWEAK_TWKFB_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${AUDIO_TWEAK_TWKAUDIO_HEADERS} ${STAGE_TWEAK_TWKAudio_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${BASE_TWEAK_TWKEXC_HEADERS} ${STAGE_TWEAK_TWKExc_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${BASE_TWEAK_TWKMATH_HEADERS} ${STAGE_TWEAK_TWKMath_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${IMAGE_TWEAK_TWKMOVIE_HEADERS} ${STAGE_TWEAK_TWKMovie_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${EXT_IMATH_INCLUDE_ROOT} ${STAGE_IMATH_HEADERS}
+  COMMAND ${CMAKE_COMMAND} -E copy ${EXT_BOOST_PROGOPTIONS_MAIN_HEADER} "${STAGE_BOOST_HEADERS}/program_options.hpp"
+  COMMAND ${CMAKE_COMMAND} -E copy ${EXT_BOOST_INCLUDE_ROOT}/config.hpp ${STAGE_BOOST_HEADERS}
```
Comment (geffrak) [inline src/lib/image/mio_ffmpeg/CMakeLists.txt:169]: can't we copy all of boost include?

**Review by geffrak — APPROVED** [review]

File: src/lib/image/mio_ffmpeg/CMakeLists.txt (hunk 2 above)
Comment (tbabiin) [reply src/lib/image/mio_ffmpeg/CMakeLists.txt:169]: Too big, I think I added around 30MB (~27MB) for the installer (uncompressed), I expect all of Boost to be at least 200MB.

**Comment by sc0rp10n-py** [comment]
After a lot of struggle, I think I was able to build the application, but I get this message now
```
Running MIO_FFMPEG Test: Copying the external build files and start CTest
cd /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg && /usr/local/Cellar/cmake/3.27.1/bin/cmake -E rm -rf /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg/test_workspace/test_build
cd /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg && /usr/local/Cellar/cmake/3.27.1/bin/cmake -E make_directory /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg/test_workspace/test_build
cd /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg && /usr/local/Cellar/cmake/3.27.1/bin/cmake -E copy_directory /Users/macos/OpenRV/_build/stage/app/RV.app/Contents/src/mio_ffmpeg /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg/test_workspace
cd /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg && /usr/local/Cellar/cmake/3.27.1/bin/cmake -E env MIO_DEP_ROOT_DIR=/Users/macos/OpenRV/_build/stage/app/RV.app/Contents /usr/local/Cellar/cmake/3.27.1/bin/ctest
Test project /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg
    Start 1: MIO_FFMPEG_EXT_BUILD
1/1 Test #1: MIO_FFMPEG_EXT_BUILD .............***Failed    0.59 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.59 sec

The following tests FAILED:
	  1 - MIO_FFMPEG_EXT_BUILD (Failed)
Errors while running CTest
Output from these tests are in: /Users/macos/OpenRV/_build/src/lib/image/mio_ffmpeg/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make[3]: *** [stage/app/RV.app/Contents/PlugIns/MovieFormats/mio_ffmpeg.dylib] Error 8
make[3]: *** Deleting file `stage/app/RV.app/Contents/PlugIns/MovieFormats/mio_ffmpeg.dylib'
make[2]: *** [src/lib/image/mio_ffmpeg/CMakeFiles/mio_ffmpeg.dir/all] Error 2
make[1]: *** [CMakeFiles/main_executable.dir/rule] Error 2
make: *** [main_executable] Error 2
```

**Pull request merged**